nums/core/array/utils.py

```diff
@@ -118,6 +118,27 @@ def can_broadcast_shapes(a_shape, b_shape):
         return False
 
 
+def broadcastable(a_shape, b_shape, a_block_shape, b_block_shape):
+    if a_shape == b_shape:
+        return a_block_shape == b_block_shape
+
+    result_shape = broadcast_shape(a_shape, b_shape)
+    if result_shape is None:
+        return False
+    if len(a_shape) == len(b_shape):
```

Comment:
What is the purpose of this check?

Reply:
I think I discussed this with Melih, as he did most of the PR. I only took over in terms of writing some tests. 

I think it's a case to broadcast block arrays when block shapes matches up? Please correct me if this is wrong @elibol. I can try to make another test specific to that functionality if needed.

Reply:
That's right. Put another way, if the two arrays have the same shape length, and the block shapes are the same, then they may be broadcasted.